src/signalingclient/DefaultSignalingClient.ts

```diff
@@ -211,6 +214,25 @@ export default class DefaultSignalingClient implements SignalingClient {
       this.sendEvent(
         new SignalingClientEvent(this, SignalingClientEventType.WebSocketClosing, null)
       );
+
+      // Continue resetting the connection even if SDK does not receive the "close" event.
+      const scheduler: TimeoutScheduler = new TimeoutScheduler(
+        DefaultSignalingClient.CLOSE_EVENT_TIMEOUT_MS
+      );
+      const handler = (event: CloseEvent): void => {
+        this.webSocket?.removeEventListener('close', handler);
+        scheduler.stop();
+        this.closeEventHandler(event);
+      };
+
+      // Remove the existing close handler to prevent SDK from opening a new connection.
+      this.webSocket?.removeEventListener('close', this.closeEventHandler);
+      this.webSocket.addEventListener('close', handler);
+      scheduler.start(() => {
+        handler(
+          new CloseEvent('close', { wasClean: false, code: 1006, reason: '', bubbles: false })
```

Comment:
Can we document what close code 1006 is and why we choose it here?

Reply:
Added a comment